src/utility/Albany_Accessors.hpp

```diff
@@ -0,0 +1,38 @@
+#ifndef ALBANY_ACCESSORS_HPP
+#define ALBANY_ACCESSORS_HPP
+
+#include "Albany_Meta_Utils.hpp"
+#include "PHAL_AlbanyTraits.hpp"
+
+
+namespace Albany
+{
+
+template<typename ST>
+struct Accessor {
+  ST& getValue () { return value; }
+  ST value;
+};
+
+template<typename EvalT>
+struct Accessors {
+  std::map<std::string,std::shared_ptr<Accessor<typename EvalT::ScalarT>>> accessors;
+};
+
+using AccessorsKeyList =
+  Albany::TypeList<PHAL::AlbanyTraits::Residual,
+                  PHAL::AlbanyTraits::Jacobian,
+                  PHAL::AlbanyTraits::Tangent,
+                  PHAL::AlbanyTraits::DistParamDeriv,
+                  PHAL::AlbanyTraits::HessianVec>;
+using AccessorsValueList =
```

Comment:
Since you are at it, you could use the `ApplyTemplate` meta util you took from EKAT, and do

```
template<typename EvalT>
using AccessorsRCP = Teuchos::RCP<Albany::Accessors<EvalT>>;

using AccessorsValueList = typename ApplyTemplate<AccessorsRCP,AccessorsKeyList>::type;
```